buildenv/jenkins/common/pipeline-functions

```diff
@@ -132,23 +132,29 @@ def build_with_slack(JOB_NAME, PARAMETERS) {
     def MESSAGE = ''
     def JOB = build job: JOB_NAME, parameters: PARAMETERS, propagate: false
 
-    if (JOB.resultIsWorseOrEqualTo('UNSTABLE')) {
-        try {
-            DOWNSTREAM_JOB_NUMBER = JOB.getNumber()
-        } catch (er) {
-            echo "Couldn't retrieve downstream failed job number"
-        }
-        try {
-            DOWNSTREAM_JOB_URL = JOB.getAbsoluteUrl()
-        } catch (err) {
-            echo "Couldn't retrieve downstream failed job url"
-        }
-        MESSAGE = "Downstream job ${JOB_NAME} did not pass. Job Number: ${DOWNSTREAM_JOB_NUMBER} Job URL: ${DOWNSTREAM_JOB_URL}"
+    try {
+        DOWNSTREAM_JOB_NUMBER = JOB.getNumber()
+    } catch (er) {
+        echo "Couldn't retrieve downstream failed job number"
+    }
+    try {
+        DOWNSTREAM_JOB_URL = JOB.getAbsoluteUrl()
+    } catch (err) {
+        echo "Couldn't retrieve downstream failed job url"
+    }
+    MESSAGE = "Downstream job ${JOB_NAME} did not pass. Job Number: ${DOWNSTREAM_JOB_NUMBER} Job URL: ${DOWNSTREAM_JOB_URL}"
+    
+    if (JOB.result == ("FAILURE")) {
         if (SLACK_CHANNEL) {
             slackSend channel: SLACK_CHANNEL, color: 'danger', message: "Failure in: ${env.JOB_NAME} #${env.BUILD_NUMBER} (<${env.BUILD_URL}|Open>)\nDownstream Job: ${JOB_NAME} #${DOWNSTREAM_JOB_NUMBER} (<${DOWNSTREAM_JOB_URL}|Open>)"
         }
         error MESSAGE
-    } 
+    } else if (JOB.result == ("UNSTABLE")) {
+    	currentBuild.result = "UNSTABLE"
+    	if (SLACK_CHANNEL) {
+            slackSend channel: SLACK_CHANNEL, color: 'warning', message: "Unstable: ${env.JOB_NAME} #${env.BUILD_NUMBER} (<${env.BUILD_URL}|Open>)\nDownstream Job: ${JOB_NAME} #${DOWNSTREAM_JOB_NUMBER} (<${DOWNSTREAM_JOB_URL}|Open>)"
+        }
```

Comment:
I think we should add an echo here, similar to the `error` above but just print a `WARNING: ... `
We can probably get rid of the `def MESSAGE = ''`since we only define/use it once.
So, `if "FAILURE; error "failed"; else if "UNSTABLE"; echo "warning: unstable"` something like that?